Rubberduck.CodeAnalysis/Inspections/Concrete/AssignmentNotUsedInspection.cs

```diff
@@ -40,8 +40,17 @@ protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
 
                 var tree = _walker.GenerateTree(parentScopeDeclaration.Context, variable);
 
-
-                nodes.AddRange(tree.GetIdentifierReferences());
+                var references = tree.GetIdentifierReferences();
+                nodes.AddRange(references.Where(r =>
+                {
+                    switch(r.Context.Parent) 
+                    {
+                        case VBAParser.SetStmtContext setStmtContext:
+                            return setStmtContext.expression().GetText() != "Nothing";
+                        default:
+                            return true;
+                    }
```

Comment:
Why the switch statement with a single case here? You can still pattern match in a simple lambda:

```
                nodes.AddRange(references.Where(r =>
                    r.Context.Parent is VBAParser.SetStmtContext setStmtContext &&
                    !setStmtContext.expression().GetText().Equals(Tokens.Nothing)));
```